src/engine.ts

```diff
@@ -288,13 +292,14 @@ export class Engine<A extends string = never, T extends Task<A> = Task<A>> {
    * @param task The current executing task.
    */
   do(task: T): void {
```

Comment:
We could change the signature to `do(task: T, macroString: string)` and pass the macro explicitly in `Engine.execute` instead of relying on the global Macro preference